File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        int aCnt = 0;
        int bCnt = 0;
        Character letter = '.';
        boolean check = false;
        
        for(int i = 0; i<s.length(); i++){
            if (check == false){
                letter = s.charAt(i);
                aCnt++;
                check = true;
                
            }
            else {
                if (s.charAt(i) == letter){
                    aCnt++;
                }
                else {
                    bCnt++;
                }
            }

            if(aCnt == bCnt){
                check = false;
                aCnt = 0;
                bCnt = 0;
                answer++;
            }
            else if(i == s.length()-1){
                answer++;
            }
     
        }
        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    i = 0
    
    while i < len(s):
        x = s[i]  # 첫 번째 글자
        x_count = 0  # x의 개수
        other_count = 0  # x가 아닌 글자의 개수
        
        # 문자열을 읽으면서 x와 x가 아닌 글자의 개수를 센다
        for j in range(i, len(s)):
            if s[j] == x:
                x_count += 1
            else:
                other_count += 1
            
            # 두 횟수가 같아지는 순간 분리
            if x_count == other_count:
                i = j + 1
                break
        else:
            # 더 이상 읽을 글자가 없을 경우 분리
            i = len(s)
        
        answer += 1 
    return answer